DEVELOPING.md

```diff
@@ -83,6 +83,54 @@ if *debugging {
 }
 ```
 
+## Git Workflow
```

Comment:
I'm a little torn here:
* On the one hand, a lot of these instructions are not specific to this repository/project, so I'm not sure we should include them here.
* On the other hand, having these instructions could help a newcomer to the project get started more quickly, especially if they are not familiar with Git/GitHub.

Maybe we could put these instructions in a dedicated file?

Independent of that, there are two general guidelines that I think we should include:
1. Please avoid force-pushing, as it can break links to commits and cause GitHub to lose track of comment threads.
2. When merging your PR, please use a squash merge. (This should be the only available option.) @PurelyApplied thoughts on squash merges? I remember we've talked about this in the past.

Independent of these concerns, I think these instructions are clear and well-written. Good job!

Reply:
I personally see very little downside to having a section for this, especially because of the reasons of getting people without a lot of Git/Github experience started quickly, like you mentioned. 

Added notes for avoiding force-pushing and using squash merge.

Reply:
That's fair. I still think this shouldn't be directly in `DEVELOPING.md`. I think a dedicated `GIT_WORKFLOW.md` file (or some other name) would be more appropriate. `DEVELOPING.md` could link to it.

I'd like someone else to weight in. @PurelyApplied, @benhxy, thoughts?

Reply:
Moved tp GIT_WORKFLOW.md